src/rules/new-url.ts

```diff
@@ -0,0 +1,64 @@
+import type { Rule } from 'eslint';
+import type * as ESTree from 'estree';
+
+const findTryStatementRange = (
+  node: ESTree.Node & Partial<Rule.NodeParentExtension>,
+): [number, number] | null => {
+  if (!node.parent) {
+    return null;
+  }
+  if (node.parent.type === 'TryStatement') {
+    /**
+     * The `try` statement should not be the only condition to determine
+     * whether the `new URL()` call is wrapped in `try-catch` block,
+     * because the following code will not throw an error in JavaScript:
+     *
+     * ```js
+     * try {
+     *   // some code that throws an error
+     * } catch {
+     *   new URL('');
+     * } finally {
+     *   new URL('');
+     * }
+     *
+     * So if node is wrapped in `catch` or `finally` block, we shoud
+     * still report the error.
+     */
+    if (node.parent?.handler === node || node.parent?.finalizer === node) {
+      return null;
+    }
+    return node.range;
+  }
+  return findTryStatementRange(node.parent);
+};
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Use try-catch to wrap new URL()',
+    },
+    fixable: null,
+    messages: {
+      shouldWrap: 'new URL() should be wrapped in try-catch',
+    },
+  },
+  create: (context: Rule.RuleContext): Rule.RuleListener => {
+    return {
+      NewExpression: node => {
+        if (node.callee.type === 'Identifier' && node.callee.name === 'URL') {
```

Comment:
There're some static methods in `URL`, for example:

```js
const blob = URL.createObjectURL(file);
URL.revokeObjectURL(blob);
```

If we enable this rule, we have to wrap up the above code in try-catch, which exceeds the scope of this rule (remember the name is `"new-url"` not `"url"`).

For the static methods, please refer to https://developer.mozilla.org/en-US/docs/Web/API/URL#static_methods

Reply:
Oh I just found it's `MemberExpression` in my code, not `NewExpression`. It's ok to judge `"Identifier"` only.